### DISCUSSION TITLE
Is it allowed to paste code from external open source libraries ?

### DISCUSSION THREAD

**Discussion by AnasImloul** [geeksblabla/blanat#49] [Q&A]
Is it permissible to copy code from other external open source projects, specifically for tasks like parsing the CSV file or finding a hashtable better than the one STL offers? I'm interested in knowing if this counts as a fraudulent attempt or not.

**Comment by chermehdi** [comment]
Two rules you should follow:

1. Keep the solution length to a minimum, we don't want to review > 5k LoC solutions
2. Ensure that you mention which code was copied as comments in the code and links to the original location/author of the code.

as long as you do both, your submission would be counted.
